app/controllers/posts_controller.rb

```diff
@@ -23,6 +25,14 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.author.decrement!(:posts_counter)
+    @post.destroy
+
+    redirect_to root_path, status: :see_other
+  end
+
```

Comment:
- [ ] Good job so far 👏 👏 As suggested by the previous reviewer, Please use the method `load_and_authorize_resource` given by the CanCan gem, at the top of the Posts_controller so that ruby will first check if the resource is authorized before running the resource 😃 It is very useful. Please read this small part in this CanCan documentation 👉  [here](https://github.com/CanCanCommunity/cancancan#controller-helpers) you will understand why you need to use it and how 😉